app/src/main/java/com/greenmonkeys/handmade/payout_denominations/CalculatePesos.java

```diff
@@ -0,0 +1,59 @@
+package com.greenmonkeys.handmade.payout_denominations;
+
+import java.util.List;
+
+public class CalculatePesos {
+    private int[] pesos = new int[]{500,200,100,50,20,10,5,2,1};
+    private int[] centavos = new int[]{50,20,10,5,1};
+
+    /**
+     * This method calculates the different payout denominations
+     * and returns a wallet containing these denominations
+     * @param totalAmount
+     * @return
+     */
+    public PesoWallet calculateSingle(double totalAmount){
+        double amount = totalAmount;
+        int[] pesoCounter = new int[9];
+
+        //do some whole peso calculations
+        for(int i = 0; i<9; i++){
+            if(amount >= pesos[i]){
+                pesoCounter[i] = (int)amount / pesos[i];
+                amount = amount - pesoCounter[i] * pesos[i];
+            }
+            else{pesoCounter[i] = 0;}
+        }
+
+        //now do some centavos calculations
+        amount*=100;
+        int[] centCounter = new int[6];
+        for(int i  = 0; i<5; i++){
+            if(amount >= centCounter[i]){
+                centCounter[i] = (int)amount / centavos[i];
+                amount = amount - centCounter[i] * centavos[i];
+            }
+            else{centCounter[i] = 0;}
+        }
+
+        return new PesoWallet(pesoCounter[0], pesoCounter[1], pesoCounter[2],
```

Comment:
PesoWallet constructor should take one (or two) arrays/maps of values. If you NEED to parse it back out into individual variables (which you shouldn't), do so in the constructor method of PesoWallet.